### PR TITLE
Varous minor changes

### PR DESCRIPTION
**Pull request by Ryan-Nolan**
- Added selected and transparent options to listbox
- Replaced the for loop around self.data.items with table.concat()
- Updated the params var and added 2 new methods for setting and getting the params
- Removed list entry that was causing a blank list item in thetextlist

### CODE REVIEW

File: smartfs.lua [hunk 1]
```diff
@@ -595,29 +604,23 @@ smartfs.element("checkbox",{
...
 		end
-                for i,value in ipairs(self.data.items) do
-                    listformspec = listformspec..value..","
-                end
-                listformspec = string.sub(listformspec, 0, -2) --removes extra ,
-                --close out the list items section
-                listformspec = listformspec..";"
-
-                --TODO support selected idx and transparency
+		print(dump(self.data.items))
```
Comment (rubenwardy) [inline smartfs.lua]: Given your earlier pr... ;)

Comment (rubenwardy) [inline smartfs.lua]: I can remove that on merge, though.

Comment (crazyR14) [inline smartfs.lua]: haha i thought i removed it, its upto you i can removeit before merge or you can do it after. what ever is best for you :package :P

**Comment by rubenwardy** [comment]
Huh. I completely forgot about this PR. Merging now. XD.

**Comment by rubenwardy** [comment]
14f15a3172ad38d485e33f94cd9e6119dd95326f

**Pull request closed without merging**